.github/workflows/nightly-upstream-snapshot-build.yml

```diff
@@ -94,19 +102,319 @@ jobs:
           build-args: "ADOT_JAVA_VERSION=${{ env.ADOT_JAVA_VERSION }}"
           context: .
           platforms: linux/amd64,linux/arm64
-          tags: |
-            ${{ env.IMAGE_TAG }}
-
-      - name: Set image name to output
-        id: imageNameOutput
-        run: echo "imageName=${{ env.IMAGE_NAME }}" >> "$GITHUB_OUTPUT"
+          tags: ${{ steps.imageNameOutput.outputs.rcImage}}
 
       - name: Upload to GitHub Actions
         uses: actions/upload-artifact@v3
         with:
           name: aws-opentelemetry-agent.jar
           path: otelagent/build/libs/aws-opentelemetry-agent-*.jar
 
+      - name: Build and push Sample-Apps without Auto-Instrumentation Agent
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: jibBuildWithoutAgent
+        env:
+          COMMIT_HASH: ${{ steps.generate_time_stamp.outputs.nowTimeTag }}
+
+  create-test-ref:
+    runs-on: ubuntu-latest
+    outputs:
+      testRef: ${{ steps.setRef.outputs.ref }}
+    steps:
+      - name: Set testRef output
+        id: setRef
+        run: echo "ref=terraform" >> $GITHUB_OUTPUT
```

Comment:
You can remove this job and have the `testRef` set as a preset environment variable instead of as an output to a step.

Reply:
Also could be a good idea to comment why we are referring to a specific branch.

Reply:
Thanks. I have reformatted now. `terraform` is main branch for test-framework.